synapse/storage/databases/main/deviceinbox.py

```diff
@@ -653,7 +653,7 @@ async def _remove_deleted_devices_from_device_inbox(
 
         def _remove_deleted_devices_from_device_inbox_txn(
             txn: LoggingTransaction,
-        ) -> int:
+        ) -> Tuple[int, bool]:
             """stream_id is not unique
             we need to use an inclusive `stream_id >= ?` clause,
             since we might not have deleted all dead device messages for the stream_id
```

Comment:
this is now somewhat misleading. We will delete all messages for a given stream id at once.